diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter1/IdentifierNaming.kt

```diff
@@ -139,6 +145,34 @@ class IdentifierNaming(configRules: List<RulesConfig>) : DiktatRule(
         return false
     }
 
+    /**
+     * method checks that identifier is correct backing field
+     */
+    private fun ASTNode.isCorrectBackingField(variableName: ASTNode): Boolean {
+        val propertyNodes = this.treeParent.getAllChildrenWithType(KtNodeTypes.PROPERTY)
+        // check that backing field name is correct
+        if (variableName.text.commonPrefixWith("_").length == 1 && variableName.text.drop(1).isLowerCamelCase()) {
+            val matchingNode = propertyNodes.find { propertyNode ->
+                val nodeType = this.getFirstChildWithType(TYPE_REFERENCE)
+                val propertyType = propertyNode.getFirstChildWithType(TYPE_REFERENCE)
+                // check that property and backing field has same type
+                val sameType = propertyType?.text == nodeType?.text
+                // check that property USER_TYPE is same as backing field NULLABLE_TYPE
+                val nodeNullableType = nodeType?.getFirstChildWithType(NULLABLE_TYPE)
+                val sameTypeWithNullable = propertyType?.getFirstChildWithType(USER_TYPE)?.text ==
+                        nodeNullableType?.getFirstChildWithType(USER_TYPE)?.text
+                val matchingNames = propertyNode.getFirstChildWithType(IDENTIFIER)?.text == variableName.text.drop(1)
+                val isPrivate = this.getFirstChildWithType(MODIFIER_LIST)?.getFirstChildWithType(PRIVATE_KEYWORD) != null
```

Comment:
I guess we should already have `isPrivate` method in some utils, is not it?

Reply:
No, we haven't